src/gui/importCIFDialog.cpp

```diff
@@ -326,84 +219,103 @@ bool ImportCIFDialog::createMoietyRemovalNETA(std::string definition)
 
 void ImportCIFDialog::on_MoietyRemoveAtomicsCheck_clicked(bool checked)
 {
-    if (checked)
-        updateFlags_.setFlag(CIFHandler::UpdateFlags::CleanMoietyRemoveAtomics);
-    else
-        updateFlags_.removeFlag(CIFHandler::UpdateFlags::CleanMoietyRemoveAtomics);
-
-    update();
+    cifHandler_.setRemoveAtomics(checked);
+    updateWidgets();
 }
 
 void ImportCIFDialog::on_MoietyRemoveWaterCheck_clicked(bool checked)
 {
-    if (checked)
-        updateFlags_.setFlag(CIFHandler::UpdateFlags::CleanMoietyRemoveWater);
-    else
-        updateFlags_.removeFlag(CIFHandler::UpdateFlags::CleanMoietyRemoveWater);
-
-    update();
+    cifHandler_.setRemoveWaterAndCoordinateOxygens(checked);
+    updateWidgets();
 }
 
-void ImportCIFDialog::on_MoietyRemoveByNETAGroup_clicked(bool checked)
+void ImportCIFDialog::on_MoietyRemoveByNETACheck_clicked(bool checked)
 {
-    if (checked)
-        updateFlags_.setFlag(CIFHandler::UpdateFlags::CleanMoietyRemoveNETA);
-    else
-        updateFlags_.removeFlag(CIFHandler::UpdateFlags::CleanMoietyRemoveNETA);
+    if (ui_.MoietyNETARemovalIndicator->state() != CheckIndicator::OKState)
+        return;
 
```

Comment:
Currently the MoietyNETARemoveFragmentsCheck and the MoietyNETARemovalEdit are disabled, I am assuming here is where they should be enabled.

```suggestion

ui_.MoietyNETARemovalEdit->setEnabled(checked);
ui_.MoietyNETARemoveFragmentsCheck->setEnabled(checked);

```

Reply:
Actually, the logic to enable/disable those controls based on the state of `ui_.MoietyNETARemoveCheck` is captured in the UI file itself with signals/slots/

Reply:
Oh, and which I see now somehow don't work.....